tools/sof-dump-status.py

```diff
@@ -82,6 +103,33 @@ def loadDMI(self):
             self.dmi[name] = line[idx:]
         pass
 
+    def loadDT(self):
+        def _loadDT(filename, is_required=False):
+            # path to file containing SoC-related information
+            filepath = "/sys/devices/soc0/" + filename
+
+            # try to read the content found at filepath
+            # if said content is not marked as required then just
+            # silence the error and print a warning message
+            try:
+                self.dt_info[filename] = read_file(filepath).strip()
+            except IOError:
+                if is_required:
+                    raise
+                else:
+                    print("[WARNING]: Unable to read content of {}.".format(filepath),
+                          file=sys.stderr)
+                    return
+
+        # read all required SoC-related information
+        try:
+            _loadDT("soc_id", is_required=True)
+        except IOError:
```

Comment:
I don't understand the purpose of the `_loadDT()` indirection... what would be wrong with a "flatter", single `loadDT()` function that does everything in a single `try` block?

Reply:
In the current implementation that would work, yes, but what if we need more SoC-related information? Wouldn't it be better to keep the internal `_loadDT`? Example:

```
# required SoC-related information
try:
    _loadDT("soc_id", is_required=True)
    _loadDT("family", is_required=True)
except IOError:
    dt_info = {}
    return

# optional SoC-related information
_loadDT("machine")
_loadDT("revision")
```